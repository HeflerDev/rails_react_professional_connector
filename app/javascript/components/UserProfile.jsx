import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import axios from 'axios';
import Appointment from './subComponents/Appointment';
import loadingGif from '../../assets/gifs/loading_2.gif';

const mapStateToProps = (state) => ({
  user: state.userReducer.userData,
  isLoggedIn: state.userReducer.isLoggedIn,
});

const ConnectedUserProfile = ({ user, isLoggedIn }) => {
  const [loaded, setLoaded] = useState(false);
  const [appointments, setAppointments] = useState([]);

  useEffect(() => {
    axios.get('/user/appointments')
      .then((res) => {
        if (res.data.status === 'fetched') {
          setAppointments(res.data.user_appointments);
        } else {
          throw new Error();
        }
      })
      .then(() => setLoaded(true))
      .catch((err) => console.log(err));
  }, []);

  const styleRow = (input) => {
    if (input % 2 === 0) {
      return 'odd';
    }
    return '';
  };

  return (isLoggedIn && loaded ? (
    <div className="stack">
      <h2>
        Hello,
        {user.username}
        {' '}
        !
      </h2>
      <h3>Your Schedule is bellow</h3>
      <div className="board center">
        {
          appointments.map((item, index) => (
            <div key={item.schedule} className={`col-10 col-m-6 ${styleRow(index)}`}>
              <Appointment data={item} userData={user} />
            </div>
          ))
        }
      </div>
    </div>
  ) : (
    <>
      <div className="queue center">
        <img src={loadingGif} alt="Loading" />
      </div>
    </>
  ));
};

ConnectedUserProfile.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string,
    id: PropTypes.number,
  }),
  isLoggedIn: PropTypes.bool.isRequired,
};

ConnectedUserProfile.defaultProps = {
  user: {
    username: null,
    id: null,
  },
};

const UserProfile = connect(mapStateToProps)(ConnectedUserProfile);

export default UserProfile;
