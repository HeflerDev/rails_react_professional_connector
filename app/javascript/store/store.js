import { createStore, combineReducers } from 'redux';
import userReducer from '../reducers/user';

const rootReducer = combineReducers({
  userReducer,
});

const store = createStore(rootReducer);

export default store;
