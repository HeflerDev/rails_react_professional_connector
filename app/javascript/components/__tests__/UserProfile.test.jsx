import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from "react-router-dom";
import store from '../../store/store';
import UserProfile from '../UserProfile';

describe('when rendering navbar', () => {
  test('it renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <UserProfile />
        </Provider>
      </BrowserRouter>
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
