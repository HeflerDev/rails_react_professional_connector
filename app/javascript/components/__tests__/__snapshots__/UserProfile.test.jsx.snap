// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering navbar it renders correctly 1`] = `"Loading..."`;
